client/components/tinymce/plugins/media/plugin.jsx

```diff
@@ -226,6 +226,12 @@ function mediaButton( editor ) {
 			if ( current.media.transient ) {
 				transients++;
 				isTransientDetected = true;
+
+				// Mark the image as a transient in upload
+				editor.dom.$( img ).toggleClass( 'is-transient', true );
```

Comment:
As per the discussion before: 
> So let's still keep the is-transient manipulation so we can experiment with alternatives. ;)

It will serve to mark the images that are currently being uploaded, so we can do some different styling for them. If there's a better way to handle transient uploads, I'll be happy to improve/change this.